From ce2235db8ef00306386fbb4ccf396989ba750f17 Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@cloudera.com>
Date: Tue, 26 Mar 2013 07:52:08 +0530
Subject: [PATCH 052/596] MR1: CLOUDERA_BUILD. Fix regression introduced by
 b5eaf8099 that makes a certain JobTracker limit
 configurable by clients if un-finalized.

Description: The JT's split meta info limits were being taken from client configs due to an earlier regression change. This fixes the bad behavior.
Reason: Customer Request
Author: Harsh J
Ref: CDH-10222
(cherry picked from commit 3d66cbd755b0ad77f35f2bc45f236b25f7854f25)
(cherry picked from commit ad79f49e3ee3a4f7b67b0cb472809a88cf84010a)
(cherry picked from commit d7b777458c90c16c9c0f8de2e6565a67a808596a)
(cherry picked from commit 9ba48cef11f23cc0fc02f29dc8345f944f266e83)
---
 .../org/apache/hadoop/mapred/JobInProgress.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/JobInProgress.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/JobInProgress.java
index ff39f57..202ac31 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/JobInProgress.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/JobInProgress.java
@@ -826,7 +826,7 @@ public Object run() throws Exception {
   TaskSplitMetaInfo[] createSplits(org.apache.hadoop.mapreduce.JobID jobId)
   throws IOException {
     TaskSplitMetaInfo[] allTaskSplitMetaInfo =
-      SplitMetaInfoReader.readSplitMetaInfo(jobId, fs, conf, jobSubmitDir);
+      SplitMetaInfoReader.readSplitMetaInfo(jobId, fs, jobtracker.getConf(), jobSubmitDir);
     return allTaskSplitMetaInfo;
   }
 
-- 
1.7.9.5

